db/migrate/20151102211125_add_slug_to_bit_core_tool.rb

```diff
@@ -0,0 +1,5 @@
+class AddSlugToBitCoreTool < ActiveRecord::Migration
+  def change
+    add_column :bit_core_tools, :slug, :string
```

Comment:
My problem with allowing it to be `NULL` is that you will have to add conditional logic and `try` anywhere that references it because you can't even assume it's a string. Also, I think it should be indexed as unique. Otherwise, you'll have no way of uniquely referencing two tools if they accidentally have the same slug.